imap_data_access/file_validation.py

```diff
@@ -68,6 +69,51 @@ def __init__(self, filename: str | Path, data_dir: Path | None = None):
         if self.error_message:
             raise self.InvalidScienceFileError(f"{self.error_message}")
 
+    @classmethod
+    def generate_from_inputs(
+        cls,
+        instrument: str,
+        level: str,
+        descriptor: str,
+        start_time: str,
+        end_time: str,
+        version: str,
+    ):
+        """Generate a filename from given inputs and return a ScienceFilePath instance.
+
+        This can be used instead of the __init__ method to make a new instance:
+        ```
+        science_file_path = ScienceFilePath.generate_from_inputs("mag", "l0", "test",
+            "20240213", "20240213", "v01-01")
+        full_path = science_file_path.construct_path()
+        ```
+
+        Parameters
+        ----------
+        descriptor : str
+            The descriptor for the filename
+        instrument : str
+            The instrument for the filename
+        level : str
+            The data level for the filename
+        start_time: str
+            The start time for the filename
+        end_time: str
+            The end time for the filename
+        version : str
+            The version of the data
+
+        Returns
+        -------
+        str
+            The generated filename
+        """
+        filename = (
```

Comment:
should we check if level is l0, then set extension to be `pkts`?

Reply:
Yes, great catch